q2_keegan/vanilla/_format.py

```diff
@@ -0,0 +1,33 @@
+import re
+
+import qiime2.plugin.model as model
+from qiime2.plugin import ValidationError
+
+from ..plugin_setup import plugin
+
+
+class VanillaBeanFmt(model.TextFileFormat):
+
+    def _validate_(self, level):
+        accepted_characters = r'[a-zA-Z\s]'
+        with self.open() as fh:
+
+            illegal_chars = {}
+
+            for line_number, line in enumerate(fh):
+                illegal_line_chars = [character for character in line if not
+                                      re.match(accepted_characters,
+                                               character)]
```

Comment:
Suggestion: this multiline list comp is hard to read, I think it would be clearer written as a "vanilla" `for` loop.